letta/server/server.py

```diff
@@ -1590,6 +1590,15 @@ def load_file_to_source(self, source_id: str, file_path: str, job_id: str, actor
         connector = DirectoryConnector(input_files=[file_path])
         num_passages, num_documents = self.load_data(user_id=source.created_by_id, source_name=source.name, connector=connector)
 
+        # update all agents who have this source attached
+        agent_ids = self.ms.list_attached_agents(source_id=source_id)
+        for agent_id in agent_ids:
+            agent = self.load_agent(agent_id=agent_id)
+            curr_passage_size = self.passage_manager.size(actor=actor, agent_id=agent_id, source_id=source_id)
+            agent.attach_source(user=actor, source_id=source_id, source_manager=self.source_manager, ms=self.ms)
+            new_passage_size = self.passage_manager.size(actor=actor, agent_id=agent_id, source_id=source_id)
+            assert new_passage_size > curr_passage_size
```

Comment:
I think this assert is causing issues on the second file upload - I think it might be because we aren't waiting for the load data to complete before re-attaching? so the passage aren't ready yet

Reply:
It turns out it's because when we call `attach_source`, there was a `page_size` limit of 100 so you could never add more than 100 passages to an agent from a source